### ISSUE TITLE
Metrics explosion since v0.35 for tracing collector component

### ISSUE THREAD

**Issue by ese** [grafana/agent#5155]
### What's wrong?

Since v0.35 tracing component is generating a massive amount of metrics due to cardinality explosion. Before this version grafana-agent /metrics endpoint generated response of around 0.7Mb and after this release is generating over 250Mb responses. I tested upon the last release v0.36.1 getting the same results.

These are the metrics causing the issue (especially becouse the label `net_sock_peer_port`). they were not being generated before using the same config:

`traces_http_server_duration_bucket`
`traces_http_server_duration_sum`
`traces_http_server_duration_count`
`traces_http_server_response_content_length_total`

### Steps to reproduce

Upgrade agent to v0.35 or later  and collect traces

### Software version

v0.36.1

### Configuration

```text
server:
        log_level: warn
    traces:
        configs:
          - batch:
                send_batch_size: 2000
                timeout: 10s
            name: default
            receivers:
                jaeger:
                    protocols:
                        grpc: null
                        thrift_binary: null
                        thrift_compact: null
                otlp:
                    protocols:
                        grpc: null
                        http: null
                zipkin: null
            service_graphs:
              enabled: false
            remote_write:
              - endpoint: tempo-distributor.tempo.svc.cluster.local:4317
                insecure: true
                sending_queue:
                    queue_size: 10000
                retry_on_failure:
                    max_elapsed_time: 5s
            scrape_configs:
              - bearer_token_file: /var/run/secrets/kubernetes.io/serviceaccount/token
                job_name: kubernetes-pods
                kubernetes_sd_configs:
                  - role: pod
                relabel_configs:
                  - action: replace
                    source_labels:
                      - __meta_kubernetes_namespace
                    target_label: namespace
                  - action: replace
                    source_labels:
                      - __meta_kubernetes_pod_name
                    target_label: pod
                  - action: replace
                    source_labels:
                      - __meta_kubernetes_pod_container_name
                    target_label: container
                  - action: replace
                    source_labels:
                      - __meta_kubernetes_pod_label_app_kubernetes_io_instance
                    target_label: release
                  - action: replace
                    source_labels:
                      - __meta_kubernetes_pod_label_app_kubernetes_io_component
                    target_label: component
                  - action: replace
                    source_labels:
                      - __meta_kubernetes_pod_label_worker
                    target_label: worker
                tls_config:
                    ca_file: /var/run/secrets/kubernetes.io/serviceaccount/ca.crt
                    insecure_skip_verify: false
```
```


### Logs

_No response_

**Comment by ese** [comment]
Could be related with https://github.com/grafana/agent/issues/4764. 

Is there a way to disable this metrics in static config?

**Comment by ptodev** [comment]
Hi @ese 👋  Thank you for your report. In static mode we don't expect such a cardinality exposion, because the `telemetry.disableHighCardinalityMetrics` feature gate is [always enabled](https://github.com/grafana/agent/blob/v0.36.0/pkg/traces/instance.go#L136-L155). This means that high cardinality labels such as `net_sock_peer_addr` should not be present.

I just tried running the main branch of the Agent with a config like this:

<details>
  <summary>Agent config</summary>

```
server:
  log_level: debug

logs:
  positions_directory: "/Users/paulintodev/Desktop/otel_test/test_log_pos_dir"
  configs:
    - name: "grafanacloud-paulintodev-logs"
      clients:
        - url: ""
          basic_auth:
            username: ""
            password: ""

traces:
  configs:
  - name: default
    receivers:
      otlp:
        protocols:
          grpc:
            endpoint: "0.0.0.0:4320"
    remote_write:
      - endpoint: localhost:4317
        insecure: true
    batch:
      timeout: 5s
      send_batch_size: 100
    automatic_logging:
      backend: "logs_instance"
      logs_instance_name: "grafanacloud-paulintodev-logs"
      roots: true
    spanmetrics:
      handler_endpoint: "localhost:8899"
      namespace: "paulin_test_"
    tail_sampling:
      policies:
        [
          {
            name: test-policy-4,
            type: probabilistic,
            probabilistic: {sampling_percentage: 100}
          },
        ]
    service_graphs:
      enabled: true
```

</details>

Indeed, I do not see high cardinality metrics. My metrics look like this - with no `net_sock_peer_addr` labels:
```
traces_rpc_server_duration_bucket{rpc_grpc_status_code="0",rpc_method="Export",rpc_service="opentelemetry.proto.collector.trace.v1.TraceService",rpc_system="grpc",traces_config="default",le="10"} 396
traces_rpc_server_duration_bucket{rpc_grpc_status_code="0",rpc_method="Export",rpc_service="opentelemetry.proto.collector.trace.v1.TraceService",rpc_system="grpc",traces_config="default",le="25"} 396
```

Would you mind telling us the precise version which you are using please? Also, it'd help if you could share a few example metrics.

It may also help to try disabling some receivers by removing them from the config. I have only tested this with "otlp" - it is possible that other receivers don't honour the feature gate, but I feel like this is unlikely.

**Comment by ese** [comment]
I tested `0.36.2` and its fixed

**Comment by jcreixell** [comment]
FYI I ran into this myself with an agent running `v0.32.1` and using the OTLP endpoint, I will upgrade and confirm it's fixed

**Comment by jcreixell** [comment]
I can still reproduce this with `v0.37.2`. The tracing bit of my config is just: 

```yaml
        traces:
          configs:
            - name: default
              receivers: # enable the receivers that you need
                otlp:
                  protocols:
                    grpc:
                    http:
              remote_write:
                - endpoint: tempo-eu-west-0.grafana.net:443
                  basic_auth:
                    username: 274733
                    password_file: /var/lib/grafana-agent/rw.key
```

and I am instrumenting a ruby/rails application with OTEL's SDK and sending the traces directly to the OTLP endpoint of the agent

**Comment by jcreixell** [comment]
I can confirm it is fixed in `v0.38.0` :raised_hands:
